### PR TITLE
Replace validators and dtos with Zod

### PR DESCRIPTION
**Pull request by alanmvarela**

### CODE REVIEW

File: src/cards/cards.controller.ts [hunk 1]
```diff
@@ -16,18 +17,19 @@ export class CardsController {
     }
 
     @Get("/:type")
-    async getCards(@ValidateCardType() type: CardType) {
-        return this.cardsService.getCards(type);
+    async getCards(@Param() typeParam: CardParamTypeDto) {
+        return this.cardsService.getCards(typeParam.type as CardType);
     }
```
Comment (AgusPk) [inline src/cards/cards.controller.ts:21]: is this casting necesary?

File: package.json [hunk 2]
```diff
@@ -9,17 +9,18 @@
...
+    "start:dev": "npm run db:startdev && nest start --watch",
     "start:debug": "nest start --debug --watch",
     "start:prod": "node dist/main",
     "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
     "test": "jest",
     "test:watch": "jest --watch",
     "test:cov": "jest --coverage",
     "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
-    "test:e2e": "jest --runInBand --config ./test/jest-e2e.json --verbose --passWithNoTests --detectOpenHandles",
+    "test:e2e": "npm run db:startdev && jest --runInBand --config ./test/jest-e2e.json --verbose --passWithNoTests --detectOpenHandles && npm run db:stopdev",
```
Comment (ulises-jeremias) [inline package.json]: in general we would like to keep the db:start in a separate step so then I can run the DB a single time and the tests any time

same on the start:dev command 😊

File: src/db/drizzle.provider.ts [hunk 3]
```diff
@@ -25,8 +25,5 @@ export class DrizzleProvider implements OnModuleInit {
     });
     this.db = drizzle(pool, { schema });
     migrate(this.db, {migrationsFolder: './src/db/migrations'});
```
Comment (ulises-jeremias) [inline src/db/drizzle.provider.ts]:
```suggestion
    await migrate(this.db, { migrationsFolder });
```

and at the top of the file add something like

```ts
import path from 'path';

const migrationsFolder = path.join(__dirname, 'migrations');
```

Comment (ulises-jeremias) [reply src/db/drizzle.provider.ts]: this is because we can't assume to have the migrations on SRC when doing the production build

**Review by ulises-jeremias — CHANGES_REQUESTED** [review]
just some changes, the rest look great!!

File: src/db/drizzle.provider.ts (hunk 3 above)
Comment (AgusPk) [reply src/db/drizzle.provider.ts]: @ulises-jeremias  #bayerLessons

File: src/cards/cards.controller.ts (hunk 1 above)
Comment (alanmvarela) [reply src/cards/cards.controller.ts:21]: Not it's not!, thanks for the heads up! It was just a left over of some alternative ways to validate the parameters i tried before arriving to the current one.

File: package.json (hunk 2 above)
Comment (alanmvarela) [reply package.json]: Fixed!

File: src/db/drizzle.provider.ts (hunk 3 above)
Comment (alanmvarela) [reply src/db/drizzle.provider.ts]: Fixed!

**Review by ulises-jeremias — APPROVED** [review]
🔥

**Pull request merged**